llvm/lib/Target/ARM/ARMAsmPrinter.cpp

```diff
@@ -695,7 +710,9 @@ void ARMAsmPrinter::emitAttributes() {
                                              DenormalMode::getPositiveZero()))
     ATS.emitAttribute(ARMBuildAttrs::ABI_FP_denormal,
                       ARMBuildAttrs::PositiveZero);
-  else if (!TM.Options.UnsafeFPMath)
+  else if (checkDenormalAttributeInconsistency(*MMI->getModule()) ||
+           checkDenormalAttributeConsistency(
+               *MMI->getModule(), "denormal-fp-math", DenormalMode::getIEEE()))
```

Comment:
How should these tags generate if `UnsafeFPMath` is replaced by denormal mode? Denormal mode is a per function attribute...

Reply:
I'd think you can just drop this, it should subsumed by the denormal-fp-math check

Reply:
Is a llvm module flag with behavior value 1 or 2 suitable here?